### ISSUE TITLE
Failed to fetch dynamically imported module

### ISSUE THREAD

**Issue by g-u-x** [strapi/strapi#21840]
### Required System information

- Node.js version: 18.20.0
- NPM version: 10.5.0
- Strapi version: 5.0.6
- Database: MySQL 8
- Operating system: Ubuntu 22.04.4
- Is your project Javascript or Typescript: TS

### Describe the bug

Strapi v5.0.0 was working great on our production server until now (self-hosted on Heztner).
After a very small update by adding a single field to an existing component, this error comes up and we can't access de admin panel anymore.
We have just upgraded to v5.0.6 and the error persists.
The admin works properly locally in my computer running both "develop" and "build"+"start" commands, same environment specs as production, besides the OS (macOS Monterey 12.7).

### Steps to reproduce the behavior

1. Update any field on a content type or component
2. Push the update to a production server
3. Run "npm run build" and "npm start"

### Expected behavior

We expect to access the admin panel.

### Screenshots

![strapi-error](https://github.com/user-attachments/assets/cd37478a-0d56-420d-98d7-853b778108b9)

**Comment by g-u-x** [comment]
UPDATE: Admin is now working back again.

As it seems to be a generic error message, in this particular case we added a new Boolean field with ‘false’ as the default value.

Admin works again after changing the default back to ‘null’.

**Comment by killthekitten** [comment]
Duplicate of #20870, also confirms that the error is a generic one

**Comment by henrik-clabs** [comment]
After updating to Node v20, this issue did not reoccur.

**Comment by killthekitten** [comment]
This issue seems to be occurring randomly and has something to do with the browser cache. I was able that the page can remain loaded and working for a few seconds, but then a new message `[vite] ✨ new dependencies optimized: @strapi/design-system` would appear and reload/break page. It seems to go away when I clean the browser cache. 

Relevant logs: 
```
[2024-11-12 19:53:51.307] http: GET /content-manager/relations/questionnaire.questionnaire-meta/vertical?id=2&pageSize=10&_q=&page=1 (37 ms) 200
[2024-11-12 19:53:51.315] http: GET /content-manager/relations/questionnaire.questionnaire-meta/2/vertical?pageSize=5&page=1 (43 ms) 200
7:53:55 PM [vite] ✨ new dependencies optimized: @strapi/design-system
7:53:55 PM [vite] ✨ optimized dependencies changed. reloading
[2024-11-12 19:53:55.995] http: GET /admin/content-manager/collection-types/api::questionnaire.questionnaire/d2kf6nmowp0wqna7syqpwn9z (92 ms) 200
The file does not exist at "/strapi/node_modules/.strapi/vite/deps/chunk-5TTMV7B3.js?v=b63c0d0b" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
The file does not exist at "/strapi/node_modules/.strapi/vite/deps/chunk-SDRGV5NY.js?v=b63c0d0b" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
The file does not exist at "/strapi/node_modules/.strapi/vite/deps/chunk-K5DLENWJ.js?v=b63c0d0b" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
The file does not exist at "/strapi/node_modules/.strapi/vite/deps/chunk-BCB6QWAC.js?v=b63c0d0b" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
The file does not exist at "/strapi/node_modules/.strapi/vite/deps/chunk-PMRURUJS.js?v=b63c0d0b" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
The file does not exist at "/strapi/node_modules/.strapi/vite/deps/chunk-TJML46GE.js?v=b63c0d0b" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
The file does not exist at "/strapi/node_modules/.strapi/vite/deps/chunk-UUYAFJHM.js?v=b63c0d0b" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
The file does not exist at "/strapi/node_modules/.strapi/vite/deps/chunk-JEQO7TY5.js?v=b63c0d0b" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
```

**Comment by killthekitten** [comment]
@g-u-x @henrik-clabs just curious, do you have any custom plugins in your project as well?

**Comment by monolithed** [comment]
2 years ago, I gave up trying to revive Strapi because of such errors, but since then, nothing has changed. I encountered the same issue, and I don't get what I did wrong. All dependencies are locked down. Yes, there's a third-party plugin, but it was integrated two weeks ago and everything was working fine. It's like a black box.

**Comment by monolithed** [comment]
```
[2024-11-28 17:12:30.826] info: Strapi started successfully
[2024-11-28 17:12:31.512] http: GET /admin/ (29 ms) 200
The file does not exist at "/www/strapi/node_modules/.strapi/vite/deps/chunk-ISNNHGRU.js?v=a22f69f4" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
^C[2024-11-28 17:12:43.989] info: Shutting down Strapi
```

Here's a fix:

```
rm -rf node_modules/
```

**Comment by WesleyPON** [comment]
We had the same issue in our production environment. In our case, the problem was resolved after disabling Cloudflare's proxy. Are others here also using a proxy? It might be worth checking.

**Comment by obafunmiso-olufemi** [comment]
Thanks @wgnahs 
I am not using cloudfare proxy.. all I am using is cloud run
Can you share your vite.config.ts
Perhaps maybe that can help

**Comment by ui-mattlucas** [comment]
> > I am getting a very similar error when making changes to the admin. Removing node_modules and reinstalling doesn't seem to be a great solution.
> > ```
> > The file does not exist at "/Users/usernamedev/projectname/node_modules/.strapi/vite/deps/chunk-XIVN3H44.js?v=65e024ea" which is in the optimize deps directory. The dependency might be incompatible with the dep optimizer. Try adding it to `optimizeDeps.exclude`.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > relevant packages :
> > "dependencies": { "@strapi/plugin-cloud": "5.11.0", "@strapi/plugin-users-permissions": "5.11.0", "@strapi/strapi": "5.11.0", "better-sqlite3": "11.3.0", "fs-extra": "^10.0.0", "mime-types": "^2.1.27", "react": "^18.0.0", "react-dom": "^18.0.0", "react-router-dom": "^6.0.0", "styled-components": "^6.0.0" }, "devDependencies": { "@types/node": "^20", "@types/react": "^18", "@types/react-dom": "^18", "typescript": "^5" },
> 
> But you can't be doing this in Production

No, I am working on this project locally trying to get it going.

**Comment by derrickmehaffy** [comment]
This error should not be possible in production unless you are running in development mode in production which we absolutely do not support.

Dynamic module loading should only ever happen in development mode, in production mode the admin is statically built and the static files are served from the backend (vite isn't even used in production mode except to build the admin panel)

**Comment by ui-mattlucas** [comment]
Thanks for the explanation @derrickmehaffy

**Comment by derrickmehaffy** [comment]
This is probably a dumb question but can you all check if your app is actually running when you see this error?

**Comment by henrik-clabs** [comment]
I did not run this in production, but after updating to Node v20, this issue did not reoccur on my test instances.

**Comment by derrickmehaffy** [comment]
> I did not run this in production, but after updating to Node v20, this issue did not reoccur on my test instances.

Now that's interesting
